mobify/hijax
# Proxy wrap

Issue: https://github.com/mobify/hijax/issues/14
## Changes
- Make Hijax a singleton so that we are not continually re-wrapping the native XHR function with the same code every time we call `new Hijax()`
## How To Test
- Run `grunt serve`
- Go to http://localhost:8888/tests/
- Inspect
- Open up the call stack of any of the `XHR finished loading: GET "http://localhost:8888/examples/response.json"`
- It is correct if we only have **one** `proxied` from `hijax.js` (Note: for the jQuery 1.3.2 test, there is no `proxied` because we proxy using `document.on('ajaxComplete')` instead)
- You can also do a `git checkout 0144b619eac66c24552403cb92a6cab373607862 src/hijax.js` to get the old version of Hijax and run the tests against that, to note that they fail 
## TODOs:
- [ ] Eng +1
- [x] Write test for feature implemented / bug fixed
- [ ] Test in an actual project...
